lib/JointspaceClient.js

```diff
@@ -469,13 +469,18 @@ class JointspaceClient {
         return this.request('audio/volume', 'GET');
     }
 
-    setVolume(volume) {
+    setVolume(volume, mute) {
+        if (typeof mute === 'undefined') {
+            mute = false;
+        }
         return this.request('audio/volume', 'POST', {
             current: volume,
-            muted: false
+            muted: mute
         });
     }
 
+    /* DOES not work on 804 - probably requires current to be set even when muting
+    use setVolume function with either static volume or get current first
     mute() {
```

Comment:
The `mute` and `unMute` methods could also be implemented like this, if the current volume is required.

```node
    mute() {
        return new Promise((resolve, reject) => {
            this.getAudioData().then((response) => {
                this.setVolume(response.current, true).then(resolve).catch(reject);
            }).catch(reject);
        })
    }
```

Not sure if the extra request would be too much overhead though.. The JointSpace api is notoriously unstable when sending multiple consecutive requests. 

Also a mute action is something that I'd expect to work as quick as possible, so maybe it's better to just completely remove these methods and keep track of the current volume at a higher level.

Reply:
This won't work because the current volume of the TV is set zo `0` if you provide muted as `true`. No matter what current value you provide. I think @PiTRiS87's initial solution is still the most elegant way to solve this.